src/agentscope/agents/react_agent.py

```diff
@@ -138,17 +147,39 @@ def reply(self, x: dict = None) -> dict:
 
             self.speak(f" ITER {_+1}, STEP 1: REASONING ".center(70, "#"))
 
-            # Generate LLM response
-            prompt = self.model.format(
-                self.memory.get_memory(),
-                Msg("system", TOOL_HINT_PROMPT, role="system"),
-            )
-
-            res = self.model(
-                prompt,
-                parse_func=parse_func,
-                max_retries=3,
-            ).raw
+            try:
+                hint_msg = Msg("system", TOOL_HINT_PROMPT, role="system")
+                self.memory.add(hint_msg)
+
+                # Generate LLM response
+                prompt = self.model.advanced_format(self.memory.get_memory())
```

Comment:
In the definition of advanced_format in dashscope_model.py, the role in the return value should be "user"?